admin/cluster_setup.py

```diff
@@ -0,0 +1,477 @@
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+"""
+Set up a flocker cluster.
+"""
+
+import sys
+import yaml
+from copy import deepcopy
+from itertools import repeat
+from json import dumps
+from pipes import quote as shell_quote
+
+from eliot import add_destination, FileDestination
+
+from treq import json_content
+
+from twisted.internet.defer import inlineCallbacks
+from twisted.python.filepath import FilePath
+from twisted.python.reflect import prefixedMethodNames
+from twisted.python.usage import Options, UsageError
+from twisted.web.http import OK
+
+from admin.acceptance import (
+    DISTRIBUTIONS,
+    LibcloudRunner,
+    ManagedRunner,
+    VagrantRunner,
+    capture_journal,
+    capture_upstart,
+    eliot_output,
+    get_trial_environment,
+)
+
+from flocker.acceptance.testtools import check_and_decode_json
+from flocker.ca import treq_with_authentication
+from flocker.common import gather_deferreds, loop_until
+from flocker.control.httpapi import REST_API_PORT
+from flocker.provision import PackageSource, Variants, CLOUD_PROVIDERS
+from flocker.acceptance.testtools import DatasetBackend
+
+
+class RunOptions(Options):
+    description = "Set up a flocker cluster."
+
+    optParameters = [
+        ['distribution', None, None,
+         'The target distribution. '
+         'One of {}.'.format(', '.join(DISTRIBUTIONS))],
+        ['provider', None, 'aws',
+         'The compute-resource provider to test against. '
+         'One of {}.'],
+        ['dataset-backend', None, 'aws',
+         'The dataset backend to test against. '
+         'One of {}'.format(', '.join(backend.name for backend
+                                      in DatasetBackend.iterconstants()))],
+        ['config-file', None, None,
+         'Configuration for compute-resource providers and dataset backends.'],
+        ['branch', None, None, 'Branch to grab packages from'],
+        ['flocker-version', None, None, 'Version of flocker to install'],
+        ['build-server', None, 'http://build.clusterhq.com/',
+         'Base URL of build server for package downloads'],
+        ['node-count', None, 2, 'Number of nodes to create (where applicable)',
+         int],
+        ['app-template', None, None,
+         'Configuration to use for each application container'],
+        ['apps-per-node', None, 1, 'Number of application containers per node',
```

Comment:
Make the default value 0. It would be reasonable to want a test cluster with no containers.